### ISSUE TITLE
Docs Site Machine Translation Technical Exploration

### ISSUE THREAD

**Issue by jpvajda** [newrelic/docs-website#2341]
We are exploring the use of machine translation for the Japanese language translation and working to define the requirements for that project it be good to start diving deeper into those requirements so we can understand the scope of work, potential approach and allow us to slot that project into our  DevEn roadmap.

### Acceptance Criteria 

- [x] set timebox
- [x] Work with @jpvajda and @jmiraNR on understanding the requirements found in G-Drive Doc called `Globalization: Machine Translation for Docs Site`
- [x] Define a technical approach and break work down with what is known 
- [x] Create an epic planning doc for this. --->  see the doc on G:Drive `Localization Epic: Machine Translation for Docs`  
- [ ] Provide a high level swag estimate of effort
- [ ] @jpvajda will work @polfliet to ensure we slot this into our product roadmap

**Comment by nr-kkenney** [comment]
just as a preface, i did research, but did not implement or actively test anything. this is all conceptual.

tl;dr -  we shouldn't need to change too much in terms of the code we have written up to currently do human translation. there are no separate APIs for MT, we only need to configure a new workflow (or update an existing one) for the project we currently use in Smartling, or create a new project. With that, we should be able to reuse most of the code / process have set up, with minimal change to accomodate machine translation.

as an example, just to illustrate how little we need to change, we could just modify our current workflow in smartling to use MT, make no changes in the repo, and we would get machine translation for the docs we send them.

In order to get human and MT translation, we need to decide on if we want a separate project for MT jobs, or if we want one project, with a conditional workflow in Smartling.

If we use a separate project, we need to:
* create the project & MT workflow in Smartling.
* store a separate projectId secret in Github.
* use the correct projectId secret for human translation vs machine translation. this should be a non-issue if we use separate workflows.

If we use a conditional workflow, we need to:
* create the conditional workflow in Smartling, decide what condition to use.
* implement how we trigger the condition.

In both cases, we should:
* create separate github workflows for machine translation. 1-1 with how we do the human translation (queue work, monitor progress, remove, etc), but with a different cron schedule since we assume this can run on a more frequent schedule.
* create a separate dynamo table, or some other logical separation (like another type on the table) for machine translation work. this is just for clarity & organization around this process. i think its easier to understand & monitor if these are as separate as possible, rather than combining them into one queue of work for example.
* add a frontmatter for pages that have been machine translated. this will be used to produce a disclaimer that the page has been machine translated. 

I think the overall MT workflow will be to use a separate project, and the steps would be:
1. identify content for MT. we can look for the lack of a translate frontmatter, or for the machine translated frontmatter mentioned above.
2. using the current code, and the new project id, send the content to Smartling. we can also use the same code for retrieval & folding changes back in.
3. after content has been machine translated, add a machine translated frontmatter.
4. when we build the site, add a machine translated disclaimer if a page contains machine translated frontmatter.

@zstix @LizBaker @roadlittledawn @jmiraNR for a first pass review 🙈  im also working on coming up with a simple diagram

**Comment by nr-kkenney** [comment]
here is a diagram capturing the current github workflows we have, with the steps outlined in red being things we would add / change for MT. everything else in black is part of the current process / unchanged.
![Untitled Diagram (2)](https://user-images.githubusercontent.com/70179215/120043605-18c3a380-bfc1-11eb-8502-9d5b5acf04e9.png)

**Comment by nr-kkenney** [comment]
after talking at length with @roadlittledawn last thursday, some of the details can be altered. in particular:
* placeholder directives may not be needed. there is already a process happening to preserve text / ignore translation. if it doesn't work for MT, then we can try look into placeholder directives.
*  we can expand the definition & usage of the `translate` frontmatter field to select documents for machine translation & identifying documents that have been machine translated. previously, a value for that field indicated we would send it to our vendor for translation, and having no value meant it wasn't translated at all. in the scenario we were thinking, a value or set of values indicates which languages are sent for human translation. the difference between the set of all languages we support, and the set in the `translate` frontmatter, then becomes what we request for machine translation.

for example, if we have:
```none
translate
  - jp
```
and the total set of languages is `[jp, kr]`, then jp is human translated, and kr is machine translated.

**Comment by jpvajda** [comment]
Up next to high level swag this with @zstix and then discuss priority with @polfliet

**Comment by zstix** [comment]
I've added a high-level estimation on each ticket. Combined they equal `17` points.

I'm closing this ticket out, now that we have wrapped up the exploration phase of this project.
